packages/blockfrost/test/blockfrostAssetProvider.test.ts

```diff
@@ -205,5 +205,56 @@ describe('blockfrostAssetProvider', () => {
       expect(BlockFrostAPI.prototype.assetsHistory).not.toHaveBeenCalled();
       expect(BlockFrostAPI.prototype.txsMetadata).not.toHaveBeenCalled();
     });
+
+    test('description and image as array', async () => {
+      BlockFrostAPI.prototype.assetsById = jest.fn().mockResolvedValue({
+        asset: '8153f8be9f05b2f32b481bbf7af877f592160b39e87f5f55c8ab035f4e46543031',
+        asset_name: '4e46543031',
+        fingerprint: 'asset1h2z87pq2tr4ksxl5nkzd2ltrrtd6vvmf5nnn46',
+        initial_mint_tx_hash: 'cb2cf241d03f0f6c523a2064b51de783db590f2871da24ce89698db2bef5fb39',
+        metadata: null,
+        mint_or_burn_count: 1,
+        onchain_metadata: {
```

Comment:
I'm wondering if this is actually the cip25 NFT metadata. If it is then we shouldn't use it as cip38 token metadata :thinking: maybe let's add some e2e test(s) (in `packages/e2e/test/blockfrost`) to fetch metadata of some token and some nft

Reply:
So I think the underlying bug is that we're trying to map cip38 (AssetInfo.tokenMetadata) into cip25 (AssetInfo.nftMetadata). We shouldn't be doing that

Reply:
So, `response.onchain_metadata `shouldn't be used to create `AssetInfo.tokenMetadata,` just `response.metadata`?

Reply:
Yes, `response.onchain_metadata` is `AssetInfo.nftMetadata`. We already have a different implementation to get `nftMetadata`, so lets just keep that for now to not break anything.